src/Microsoft.Health.Fhir.Shared.Core/Features/Search/BundleFactory.cs

```diff
@@ -61,14 +60,25 @@ public ResourceElement CreateHistoryBundle(SearchResult result)
             return CreateBundle(result, Bundle.BundleType.History, r =>
             {
                 var resource = new RawBundleEntryComponent(r.Resource);
-                var hasVerb = Enum.TryParse(r.Resource.Request?.Method, true, out Bundle.HTTPVerb httpVerb);
 
+                var hasVerb = Enum.TryParse(r.Resource.Request?.Method, true, out Bundle.HTTPVerb httpVerb);
+#if Stu3
+                // STU3 doesn't have PATCH verb, so let's map it PUT.
+                if (!hasVerb && string.Equals("PATCH", r.Resource.Request?.Method, StringComparison.OrdinalIgnoreCase))
+                {
+                    hasVerb = true;
+                    httpVerb = Bundle.HTTPVerb.PUT;
+                }
+#endif
                 resource.FullUrlElement = new FhirUri(_urlResolver.ResolveResourceWrapperUrl(r.Resource, true));
-                resource.Request = new Bundle.RequestComponent
+                if (hasVerb)
```

Comment:
Should we throw an error if hasVerb is false, or will be it caught in the switch case below every time?

Reply:
Switch statement wouldn't throw exception because it's out parameter and it must be filled with something and for enums it's probably first thing in enum, which is Http.Get.

I'm reluctant to throw, since I'm not sure in which cases we don't have method, so I'll just put creation of response and request under if statement.